context/app/static/js/components/detailPage/visualization/VisualizationTracker/hooks.ts

```diff
@@ -1,90 +1,102 @@
-import React, { useCallback, useRef } from 'react';
+import React, { useMemo, useRef } from 'react';
 import useTrackMount from 'js/hooks/useTrackMount';
-import { VitessceInteraction } from './types';
-import {
-  getLastInteraction,
-  getNearestIdentifier,
-  handleKeyPress,
-  modifierKeys,
-  mouseButtonMap,
-  pushVitessceEvent,
-} from './utils';
+import { useFlaskDataContext } from 'js/components/Contexts';
+import { useEventCallback } from '@mui/material';
+import { trackEvent } from 'js/helpers/trackers';
+import { getNearestIdentifier, modifierKeys, mouseButtonMap } from './utils';
 
-type FocusAndMouseHandler = React.MouseEventHandler<HTMLSpanElement> & React.FocusEventHandler<HTMLSpanElement>;
+export function useVitessceEventMetadata() {
+  const location = window.location.href;
+  const flaskData = useFlaskDataContext();
 
-export const useVisualizationTracker = () => {
-  // Track when the visualization is first mounted
-  useTrackMount('Visualization', 'Vitessce Mounted');
-  // Create an object to track Vitessce interactions
-  // This list is populated with a summary of every interaction a user has with the visualization
-  const vitessceInteraction = useRef<VitessceInteraction>([]);
+  let category = 'Unknown';
+  let name = location; // Fall back to the full URL if we can't find a better name
```

Comment:
Can we use case/switch or similar to avoid the `let`?

Reply:
I ran into this eslint rule with the switch approach: https://eslint.org/docs/latest/rules/no-case-declarations

If I remember correctly, we'd need to return the category/name object separately in each switch statement without separately instantiating the variables with `const`.

Reply:
Ah gotcha, this is actually ringing a bell and I think I've dealt with it before. 

Would it make sense to move to a function which returns an object with category and name to be destructured? Sorry if this is just tedious, feel free to ignore!

Reply:
Good call on the helper function, that was a very good way to clean up the logic a bit (and make that individual piece testable).